src/interpreter/initialization.rs

```diff
@@ -38,26 +38,39 @@ where
         self.push_stack(tx.id().as_ref())
             .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
 
-        // Set initial unused balances
-        let free_balances = Self::initial_free_balances(&tx)?;
-        for (asset_id, amount) in free_balances.iter().sorted_by_key(|i| i.0) {
-            // push asset ID
-            self.push_stack(asset_id.as_ref())
-                .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
-            // stack position
-            let asset_id_offset = self.registers[REG_SSP] as usize;
-            self.unused_balance_index.insert(*asset_id, asset_id_offset);
-            // push spendable amount
-            self.push_stack(&amount.to_be_bytes())
-                .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
-        }
+        let predicate = !tx.is_script();
```

Comment:
this doesn't make sense, predicate behavior is totally unrelated to the transaction type

Reply:
I didn't review that part yet, its wip